src/Jackalope/Query/Row.php

```diff
@@ -179,6 +179,12 @@ public function getValue($columnName)
      */
     public function getNode($selectorName = null)
     {
+        $path = $this->getPath($selectorName);
+        if (!$path) {
+            // handle outer joins
+            return null;
```

Comment:
this is the behaviour in jackrabbit.

there is a slight risk for a BC break, if the user relied on the exception being thrown in this case.